pkg/services/auth/authimpl/external_session_store.go

```diff
@@ -0,0 +1,244 @@
+package authimpl
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+
+	"github.com/grafana/grafana/pkg/infra/db"
+	"github.com/grafana/grafana/pkg/infra/tracing"
+	"github.com/grafana/grafana/pkg/services/auth"
+	"github.com/grafana/grafana/pkg/services/secrets"
```

Comment:
Since you are using the service `secrets` from Grafana, I think we also have to add a migration for the DB secrets here: https://github.com/grafana/grafana/blob/master/pkg/services/secrets/migrator/migrator.go. This can be included in a follow up PR though.

Reply:
Thanks, done.